### ISSUE TITLE
[BUG] Single column slicing with loc returns a DataFrame not a Series

### ISSUE THREAD

**Issue by beckernick** [rapidsai/cudf#1513]
**Describe the bug**
Single column slicing with `loc` returns a DataFrame not a Series.

**Steps/Code to reproduce bug**
```python
import cudf
import pandas as pd

pdf = pd.DataFrame({'x': [1, 2, 3, 4, 5, 6, 7, 8]})
gdf = cudf.from_pandas(pdf)
print(type(pdf.loc[0:1, 'x']))
print(type(gdf.loc[0:1, 'x']))
<class 'pandas.core.series.Series'>
<class 'cudf.dataframe.dataframe.DataFrame'>
```

**Expected behavior**
I expect this to be consistent with pandas and return a Series, not a DataFrame.

**Additional context**
This bug partially blocks cumulative aggregation operations with dask-cudf.

**Comment by beckernick** [comment]
This is the [line of code](https://github.com/rapidsai/cudf/blob/branch-0.7/python/cudf/dataframe/dataframe.py#L2762). We only create a Series if the row_label is not None, which only occurs rather than also if a slice occurs. Our return type should probably not based on whether we take a single row or a slice of rows, but based on the two-way interaction of row and column indexing. Something like:

1. single row + single column -> scalar
2. single row + multiple columns -> series
3. slice + single column -> series
4. slice + multiple columns -> dataframe

Pandas actually doesn't quite do this exclusively. They also return a dataframe for slice + list of a single column. So if the row indexer is a slice and the column indexer is a list we can return a dataframe.
```python
type(pdf.loc[0:1, ['x']])
pandas.core.frame.DataFrame
```

They also return a series for a single row + list of single column:

```python
type(pdf.loc[0, ['x']])
pandas.core.series.Series
```

So essentially, they "upcast" from scalar -> and series -> dataframe if the column indexer argument is a list for the respective row indexer (single row and slice).

**Comment by beckernick** [comment]
Resolved by #1622.
